### PR TITLE
fix(compiler): cannot access struct fields

### LINKED ISSUE #1604: Compile error when accessing Struct attribute
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

I tried to access a attribute of a struct.
example
```js
bring cloud;

let x = cloud.BucketProps {
  public: true
};
let y = x.public; // error
```

or in a inflight 

```js
let handlerPost = inflight (message: cloud.ApiRequest): cloud.ApiResponse => {
  let request = cloud.ApiRequest {
   method: "GET",
   path: "/hello/world",
  };

  let resp = cloud.ApiResponse {
    body: {
      input:request.path, // error
    },
    status: 200,
    headers: {
      "Content-Type": "text/plain",
    },
  };
  return resp;
};
```

### I expected this:

The struct attribute could be accessed and the compiler does not throw an error.

### Instead, this happened:

When compiling the wing app the compiler complains with an error 
```
Error at ../../examples/tests/valid/api.w:34:13 | Expression must be a class or resource instance to access property "path", instead found type "ApiRequest"
Error at ../../examples/tests/valid/api.w:34:13 | Expression must be a class or resource instance to access property "path", instead found type "ApiRequest"
```

### Component:

Compiler

### Wing Version:

0.5.58

### Node.js Version:

v18.13.0

### Platform(s):

_No response_

### Anything else?

https://winglang.slack.com/archives/C048QCN2XLJ/p1676920474912649

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #1604

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/valid/structs.w [hunk 1]
```diff
@@ -21,6 +21,9 @@ let y = B {
   }
 };
 
+// successful struct access
+assert(x.field0 == "Sup");
+
```
Comment (hasanaburayyan) [inline examples/tests/valid/structs.w:26]: There are 2 commented out tests in this file with TODO: based on this card. Can we uncomment those?

Comment (Chriscbr) [reply examples/tests/valid/structs.w:26]: Oops, totally missed those. Looking now

Comment (Chriscbr) [reply examples/tests/valid/structs.w]: Fixed

**Review by eladb — COMMENTED** [review]
Nice

**Pull request merged**